websites/P/Perplexity/presence.ts

```diff
@@ -0,0 +1,87 @@
+const presence = new Presence({
+		clientId: "893871116945350726",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	threadExportRegex = /([a-f\d]{8}(-[a-f\d]{4}){3}-[a-f\d]{12})/i;
+let recentSearchQuery: string = null;
+
+presence.on("UpdateData", async () => {
+	const { href, pathname } = document.location,
+		[privacy, hideButtons, iconType] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("hideButtons"),
+			presence.getSetting<number>("iconType"),
+		]),
+		currentSearch = new URL(href).searchParams.get("q"),
+		pageInput = document.querySelector<HTMLInputElement>(".md\\:mb-md"),
+		currentPage = pathname.split("/")[1],
+		largeImage =
+			["https://i.imgur.com/gdCcljx.png", "https://i.imgur.com/xp4o0wQ.png"][
+				iconType
+			] || "https://i.imgur.com/gdCcljx.png",
+		presenceData: PresenceData = {
+			largeImageKey: largeImage,
+			startTimestamp: browsingTimestamp,
+		};
+	if (currentSearch && currentSearch !== recentSearchQuery)
+		recentSearchQuery = currentSearch;
+
+	presenceData.details = "Home";
+
+	if (!currentPage) return presence.setActivity(presenceData);
+
+	switch (currentPage.toLowerCase()) {
+		case "search":
+			presenceData.details = privacy ? "Searching" : "Searching for";
+			presenceData.smallImageKey = "https://i.imgur.com/wYVlwJX.png";
```

Comment:
Use standardized search asset
```suggestion
			presenceData.smallImageKey = Assets.Search;
```